src/views/MarketPlace/components/AdminPage/index.jsx

```diff
@@ -63,13 +68,71 @@ const AdminMarketplaceThreads = () => {
     fetchThreads();
   }, [page, rowsPerPage]);
 
+  useEffect(() => {
+    const trySearchById = async () => {
+      if (!searchTrigger) return;
+
+      setLoading(true);
+      try {
+        const pageSize = 100;
+        let currentPage = 1;
+        let hasMore = true;
+
+        while (hasMore) {
+          const threadsPage = await marketplaceService.getAdminThreads({
+            page: currentPage,
+            pageSize,
+          });
+
+          for (const thread of threadsPage) {
+            const history = await marketplaceService.getThreadEditHistory(thread.ThreadId);
+            const match = history.find((h) => h.Id.toString() === searchTrigger);
+            if (match) {
+              setEditHistory(history);
+              setIsModalOpen(true);
+              setSearchQuery('');
+              setSearchTrigger('');
+              setLoading(false);
+              return;
+            }
+          }
+
+          hasMore = threadsPage.length === pageSize;
+          currentPage++;
```

Comment:
This seems like more work than is necessary to find a specific database object by Id. Can you not just create an API route that gets a specific ThreadEditHistory object by Id? Even if you can't query the database by id directly (although I suspect you definitely can), you should at least move this nested searching logic into the API where it can be performed much more efficiently and not hold up the rendering on the front end.